### PR TITLE
Add header

### PR DESCRIPTION
**Pull request by yuya-minamide**
#### Code changes
- added BurgerNavMenu components
- added Header components
- added HeaderLogo components
- added NavContent components
- added NavMenu components
- fixed common style of global.scss

#### Issues affected
- #20

## Screenshots and/or video
header-web
header-mobail
header-mobail-open

## Testing Plan
- Check the design responsibility with the Google Developer tools

#### Basics
- [x] Local QA is complete
- [x] No debugging `console.log()` statements
- [x] Commented-out code removed
- [x] Unused code removed (imports, functions, styles, etc - if it's not used anywhere, it's not in this PR)

### CODE REVIEW

File: src/components/common/Header/HeaderLogo.jsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Image from 'next/image'
+import Logo from '../../../../public/icons/header-logo.png'
```
Comment (Soma-dev0808) [inline src/components/common/Header/HeaderLogo.jsx]: You can access to the asset directly /icons/<image-file-name>
https://nextjs.org/docs/basic-features/static-file-serving

File: src/components/common/Header/BurgerNavMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import styles from '../../../styles/common/Header/BurgerNavMenu.module.scss'
```
Comment (Soma-dev0808) [inline src/components/common/Header/BurgerNavMenu.jsx]: Please use absolute path when you import modules or components. `@/styles/.......`.
If the components are in the same directly or not too far from current location(./), it's okay though.

File: src/components/common/Header/BurgerNavMenu.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import { NavContent } from '../../index'
```
Comment (Soma-dev0808) [inline src/components/common/Header/BurgerNavMenu.jsx:1]: absolute path

File: src/components/common/Header/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import styles from '../../../styles/common/Header/Header.module.scss'
```
Comment (Soma-dev0808) [inline src/components/common/Header/Header.jsx]: absolute path

File: src/components/common/Header/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import { HeaderLogo, NavMenu, BurgerNavMenu } from '../../index'
+
+const Header = () => {
```
Comment (Soma-dev0808) [inline src/components/common/Header/Header.jsx]: Please separate UI component and Container component, then put the business logics in Container component.

File: src/components/common/Header/HeaderLogo.jsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import styles from '../../../styles/common/Header/HeaderLogo.module.scss'
```
Comment (Soma-dev0808) [inline src/components/common/Header/HeaderLogo.jsx]: absolute path.

File: src/components/common/Header/NavContent.jsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import styles from '../../../styles/common/Header/NavContent.module.scss'
```
Comment (Soma-dev0808) [inline src/components/common/Header/NavContent.jsx]: absolute paht

File: src/components/common/Header/NavMenu.jsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import styles from '../../../styles/common/Header/NavMenu.module.scss'
```
Comment (Soma-dev0808) [inline src/components/common/Header/NavMenu.jsx]: absolute path.

File: src/pages/index.js [hunk 9]
```diff
@@ -1,4 +1,5 @@
 import Head from 'next/head'
+import { Header } from '../components/index'
```
Comment (Soma-dev0808) [inline src/pages/index.js]: no need to specify index file. ../components

File: src/styles/global.scss [hunk 10]
```diff
@@ -11,6 +11,7 @@
   color: #3d3935;
   --main-color: #237a57;
   --sub-color: #f2f0eac8;
+  --transition: all 400ms ease;
```
Comment (Soma-dev0808) [inline src/styles/global.scss]: It's okay for now, but if you are using sass, please use variables and put them in styles/_variables.scss.

**Review by Soma-dev0808 — CHANGES_REQUESTED** [review]

File: src/components/common/Header/HeaderLogo.jsx (hunk 1 above)
Comment (yuya-minamide) [reply src/components/common/Header/HeaderLogo.jsx]: Thanks for good advice!
I fixed it👍

File: src/components/common/Header/Header.jsx (hunk 5 above)
Comment (yuya-minamide) [reply src/components/common/Header/Header.jsx]: I separated logic code from no-logic code in Header.jsx👍

File: src/styles/global.scss (hunk 10 above)
Comment (yuya-minamide) [reply src/styles/global.scss]: I added common style to _variables

**Review by Soma-dev0808 — APPROVED** [review]

**Review by HidemichiShimura — APPROVED** [review]
lgtm 👍

**Pull request merged**